src/caches/beliefpropagationcache.jl

```diff
@@ -157,6 +157,15 @@ function environment(bp_cache::BeliefPropagationCache, verts::Vector)
   return vcat(messages, central_tensors)
 end
 
+function factors(bp_cache::BeliefPropagationCache, vertices::Vector)
+  tn = tensornetwork(bp_cache)
+  return ITensor[tn[vertex] for vertex in vertices]
+end
```

Comment:
```suggestion
function factors(bp_cache::BeliefPropagationCache, vertices)
  tn = tensornetwork(bp_cache)
  return map(vertex -> tn[vertex], vertices)
end
```
I generally like using `map` better since it is a bit more generic, for example it preserves the type of the collection being input over, for example it outputs a `Dictionary` if `vertices` are stored in a `Dictionary`.